toolchain/valgrind.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Copyright: 2017 IBM
+# Author: Harish S <harish@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import build
+from avocado.utils import archive
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Valgrind(Test):
+    '''
+    Valgrind is an instrumentation framework for building dynamic analysis
+    tools. Valgrind tools can automatically detect many memory management
+    and threading bugs, and profile your programs in detail
+
+    The Test fetches the recent valgrind tar and runs the tests in it
+    '''
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.failures = []
+        deps = ['gcc', 'make']
+
+        for package in deps:
```

Comment:
please use like this way 
`
for package in ['gcc', 'make']:`  as want to eliminate extra variable